### DISCUSSION TITLE
Timestamp Tags for a Project

### DISCUSSION THREAD

**Discussion by pataiadam** [DependencyTrack/dependency-track#3994] [Q&A]
Hi, 

I came across a comment in issue [#695](https://github.com/DependencyTrack/dependency-track/issues/695) where it was mentioned that eventually, projects would have tags associated with timestamps for identification and display on charts. Is this feature currently implemented? I do not see it supported in the latest version. If not, is it still planned for a future release?

Explanation:
Our company is currently deciding whether to create separate projects for different environments (dev/stage/prod) or to use a single project and tag SBOM uploads with the appropriate environment name using the Jenkins publisher plugin. 
I personally prefer the latter, but unfortunately, tags do not appear on the charts, and I also cannot filter well within the project, or I might not be seeing how to do it correctly.

Thank you!

**Comment by nscuro** [comment]
AFAIK this has not been worked on, and there doesn't seem to be any open issue requesting it.

You can help by raising an enhancement request, and laying out your use case.

Personally, I am not really sure what was even meant with adding timestamps to tags, or how that feature is supposed to behave. So please be as specific as possible when describing your use case.

> I also cannot filter well within the project, or I might not be seeing how to do it correctly

Tags are considered when entering a value in the search field of the projects table. The corresponding REST endpoints also have means of filtering by tag.

**Comment by pataiadam** [comment]
![image](https://github.com/user-attachments/assets/29c631fd-32e8-48d5-b246-25eae3211d0c)

On the UI, what I mean by "adding timestamps to tags" is that when we upload a BOM to a project, I would like to be able to tag that specific BOM with something (like: `1.2.3` or `production`). This would allow us to track it accurately on the timeline within the charts.

For example, we could add a 'master' tag to BOMs generated only on the master branch, or if a new release is created, we could tag it with 'v1.2.3'. This way, we could follow the progress and changes more effectively.

Would this make sense? IDK Im just curious about what the best practice would be for the use case mentioned above:

> Our company is currently deciding whether to create separate projects for different environments (dev/stage/prod) or to use a single project and tag SBOM uploads with the appropriate environment name using the Jenkins publisher plugin.

I don't think it would be efficient to create a new project for every release version or even one for each environment.

**Comment by pataiadam** [comment]
Any updates?
